src/igmp.rs

```diff
@@ -0,0 +1,72 @@
+use core::mem;
+
+use crate::{getter_be, setter_be};
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+#[cfg_attr(feature = "serde", derive(::serde::Serialize, ::serde::Deserialize))]
+pub struct IGMPv2Hdr {
+    /// Message type
+    pub message_type: u8,
+
+    /// Maximum Response Time
+    pub max_response_time: u8,
+
+    /// Checksum
+    pub checksum: [u8; 2],
+
+    /// Group Address
+    pub group_address: [u8; 4],
+}
+
+impl IGMPv2Hdr {
+    /// The size of the Sctp header in bytes (8 bytes).
+    pub const LEN: usize = mem::size_of::<IGMPv2Hdr>();
+
+    #[inline]
+    pub fn checksum(&self) -> u16 {
+        // SAFETY: Pointer arithmetic in bounds of the struct.
+        unsafe { getter_be!(self, checksum, u16) }
+    }
+
+    #[inline]
+    pub fn group_address(&self) -> u32 {
+        // SAFETY: Pointer arithmetic in bounds of the struct.
+        unsafe { getter_be!(self, group_address, u32) }
+    }
+
+    #[inline]
+    pub fn set_message_type(&mut self, message_type: u8) {
+        // SAFETY: Pointer arithmetic in bounds of the struct.
+        unsafe { setter_be!(self, message_type, message_type) }
```

Comment:
Why not just use `self.message_type = message_type;` ?

Reply:
Done